legion/operator/pkg/utils/docker.go

```diff
@@ -0,0 +1,116 @@
+//
+//    Copyright 2019 EPAM Systems
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/docker/distribution/manifest/schema1"
+	"github.com/heroku/docker-registry-client/registry"
+	"github.com/legion-platform/legion/legion/operator/pkg/legion"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+)
+
+var (
+	dockerImageRegexp = regexp.MustCompile(`(.*?)/([\w\-/]+):([\-.\w]+)`)
+)
+
+type imageAttrs struct {
+	host string
+	name string
+	tag  string
+}
+
+type LabelExtractor func(image string) (map[string]string, error)
+
+func parseImage(image string) (*imageAttrs, error) {
+	submatch := dockerImageRegexp.FindStringSubmatch(image)
+	host, name, tag := submatch[1], submatch[2], submatch[3]
+
+	fmt.Println(fmt.Sprintf("Extract [host=%s,name=%s,tag=%s] from %s image", host, name, tag, image))
+
+	if host == "" {
+		return nil, errors.New("Host is empty")
+	}
+	if name == "" {
+		return nil, errors.New("Name is empty")
+	}
+	if tag == "" {
+		return nil, errors.New("Tag is empty")
+	}
+
+	return &imageAttrs{host: fmt.Sprintf("https://%s", host), name: name, tag: tag}, nil
```

Comment:
What with non-HTTPS registries? Also missed in docs.

Reply:
I recommend not to support docker registry without TLS. Because you can't deploy it in kubernetes cluster without modification a docker config in worker nodes.